### PR TITLE
[#2861] Adapt transformations to work with N dataset-versions

### PR DESCRIPTION
**Pull request by tangrammer**
- [ ] **Update release notes if necessary**
### TODO:
- [x] to have backward compatibility we need derivation column uniqueness by the dataset. Also, we need to have unique column names in the visualization/aggregation side.
- [x] transformations are spread in several dataset-version-tables depending on the transformation namespace value

#2861

https://github.com/akvo/akvo-lumen/issues/2804#issuecomment-680838633

`We decided that transformation between "regular" data groups should be possible, as they are today. But for transformations that use more than one column and one of the columns is from a repeated group, the user can only select the 2nd column from that same repeated group. We even have some work done to support this behind the feature flag: #2741
`

### CODE REVIEW

File: backend/src/akvo/lumen/endpoint/split_column.clj [hunk 1]
```diff
@@ -18,14 +18,15 @@
...
-    (let [query           (json/parse-string (get query-params "query") keyword)
-          tenant-conn     (p/connection tenant-manager tenant)
-          dataset-version (db.transformation/latest-dataset-version-by-dataset-id tenant-conn {:dataset-id dataset-id})
-          sql-query       {:table-name  (:table-name dataset-version)
-                           :column-name (:columnName query)
-                           :limit       (str (:limit query "200"))}
-          values          (map (comp str (keyword (:columnName query)))
-                               (db.transformation/select-random-column-data tenant-conn sql-query))
+    (let [query            (json/parse-string (get query-params "query") keyword)
+          namespace        (:namespace query "main")
```
Comment (tangrammer) [inline backend/src/akvo/lumen/endpoint/split_column.clj:22]: we need to update client-side to send `namespace` value

File: backend/src/akvo/lumen/lib/transformation.clj [hunk 2]
```diff
@@ -51,7 +51,7 @@
             :transformation (engine/execute-transformation tx-deps dataset-id job-execution-id (:transformation command))
             :undo (engine/execute-undo tx-deps dataset-id job-execution-id)))
         (db.job-execution/update-successful-job-execution tx-conn {:id job-execution-id}))
-      (let [dsv (db.transformation/latest-dataset-version-by-dataset-id tenant-conn {:dataset-id dataset-id})]
+      (doseq [dsv (db.transformation/latest-dataset-version-by-dataset-id tenant-conn {:dataset-id dataset-id})]
```
Comment (tangrammer) [inline backend/src/akvo/lumen/lib/transformation.clj:54]: theoretically, we only need to vacuum the table that we update not all of the dataset-version tables

File: backend/src/akvo/lumen/lib/transformation/rename_column.clj [hunk 3]
```diff
@@ -12,13 +12,12 @@
   [op-spec]
   (and (util/valid-column-name? (col-name op-spec))
        (string? (new-col-title op-spec))))
-
+;; TODO this affects on how transformations and columns in a dataset_version are related :/
```
Comment (tangrammer) [inline backend/src/akvo/lumen/lib/transformation/rename_column.clj:15]: TODO this affects on how transformations and columns in a dataset_version are related :/

File: backend/src/akvo/lumen/endpoint/split_column.clj (hunk 1 above)
Comment (tangrammer) [reply backend/src/akvo/lumen/endpoint/split_column.clj:22]: https://github.com/akvo/akvo-lumen/pull/2874

Comment (tangrammer) [reply backend/src/akvo/lumen/endpoint/split_column.clj:22]: https://github.com/akvo/akvo-lumen/pull/2874/commits/da8a88ad14da9ef07fa92eb2fab5dd0d971bfdbf

File: backend/src/akvo/lumen/lib/transformation.clj (hunk 2 above)
Comment (tangrammer) [reply backend/src/akvo/lumen/lib/transformation.clj]: WIP https://github.com/akvo/akvo-lumen/pull/2870/commits/bde31eb21ed333fb537cc44ee01695b8e013df6f

File: backend/src/akvo/lumen/lib/transformation/combine.sql [hunk 4]
```diff
@@ -1,3 +1,19 @@
 -- :name combine-columns :!
 UPDATE :i:table-name
 SET :i:new-column-name = concat (:i:first-column, :separator, :i:second-column)
+
```
Comment (tangrammer) [inline backend/src/akvo/lumen/lib/transformation/combine.sql:4]: example on updating table with several table columns

**Review by kardan — APPROVED** [review]
LGTM

**Pull request merged**